test/test_translator.py

```diff
@@ -138,8 +136,73 @@ def test_parse_polyline() :
         ]
     assert  test_coordinates == valid_coordinates
 
+def test_parse_polyline_empty_string() :
+
+    test_polyline= None
+    test_coordinates=icone_translator.parse_polyline(test_polyline)
+    valid_coordinates= []
+    assert  test_coordinates == valid_coordinates
+
+def test_parse_polyline_invalid_data() :
+    test_polyline= 'invalid' 
+    test_coordinates=icone_translator.parse_polyline(test_polyline)
+    valid_coordinates= []
+    assert  test_coordinates == valid_coordinates
 
-def test_get_road_direction():
+def test_parse_polyline_invalid_coordinates() :
```

Comment:
Rather than `assert True` or `assert False` here we should be using built-in testing tools to check this exception for us:
```
def test_parse_polyline_invalid_coordinates():
    test_polyline = 'a,b,c,d'
    with pytest.raises(RuntimeError) as runtimeErr:
        icone_translator.parse_polyline(test_polyline)
    assert "Failed to parse polyline data." in str(runtimeErr.value)
```
Here we import pytest, then we can use it to verify the exception is raised. Then we can even go so far as to assert the exception message is what we expected it to be.

Alternatively, we could wrap these tests in a class and pass that class "unittest.TestCase" then the test method can take in "self" and call "self.assertRaises":
```
class TestTranslator(unittest.TestCase):

    def test_parse_polyline_invalid_coordinates(self):
        test_polyline = 'a,b,c,d'
        with self.assertRaises(RuntimeError) as runtimeErr:
            icone_translator.parse_polyline(test_polyline)
```

This method has a big benefit of the ".assertRaises" which is asserting the exception did in fact fire off.

Reply:
i updated the code and i think i like this concept.